src/iidm/Substation.cpp

```diff
@@ -124,6 +128,47 @@ VoltageLevelAdder Substation::newVoltageLevel() {
     return VoltageLevelAdder(*this);
 }
 
+void Substation::remove() {
+    Substations::checkRemovability(*this);
+
+    std::vector<std::reference_wrapper<VoltageLevel>> voltageLevels;
+    for (VoltageLevel& vl : m_voltageLevels) {
+        voltageLevels.emplace_back(vl);
+    }
+
+    for (VoltageLevel& vl : voltageLevels | boost::adaptors::reversed) {
+        std::vector<std::reference_wrapper<Connectable>> connectables;
+        for (Connectable& connectable : vl.getConnectables()) {
+            connectables.emplace_back(std::ref(connectable));
+        }
+        // Remove all branches, transformers and HVDC lines
+        for (Connectable& connectable : connectables | boost::adaptors::reversed) {
+            ConnectableType type = connectable.getType();
+            if (VoltageLevels::MULTIPLE_TERMINALS_CONNECTABLE_TYPES.find(type) != VoltageLevels::MULTIPLE_TERMINALS_CONNECTABLE_TYPES.end()) {
+                connectable.remove();
+            } else if (type == ConnectableType::HVDC_CONVERTER_STATION) {
+                HvdcLine& hvdcLine = getNetwork().getHvdcLine(dynamic_cast<HvdcConverterStation&>(connectable));
+                hvdcLine.remove();
```

Comment:
This is not OK: consider that the HvdcLine as already been removed, this will throw an exception whereas it's totally valid to have a converter station that is not attached to a HvdcLine. You should use `findHvdcLine` and check if the HvdcLine has been found